server.py

```diff
@@ -247,10 +247,18 @@ def get_data_from_neighbors():
         request.addErrback(sync_error)
     return
 
+def delete_expired_data():
+    logger.info("Expiring data")
+    contacts.delete_expired_data()
+    return
 
 if 0 != len(servers):
-    l = task.LoopingCall(get_data_from_neighbors)
-    l.start(int(config.get('neighbor_sync_period', 600.0)))
+    l1 = task.LoopingCall(get_data_from_neighbors)
+    # TODO-DAN Shouldnt this be getfloat ?
+    l1.start(int(config.get('neighbor_sync_period', 600.0)))
+
+l2 = task.LoopingCall(delete_expired_data)
```

Comment:
this can cause problems since we may be marking json files that we want read in a thread during a scan or sync, that might get deleted here.  We are also touching the file system here during the main thread.  I was going to work on this today and will address by:
1) in the main thread/reactor move the data that should be expired from the sorted_list to to another list and update the in memory data structure
2) periodically run the a THREAD to delete the files

Reply:
Can you parameterise it then - provide a delete function in the data structures, the more we change this and the more different places I see similar code, the more I'm worried about breaking something when I touch it.

Reply:
Can you parameterise it then - provide a delete function in the data structures, the more we change this and the more different places I see similar code, the more I'm worried about breaking something when I touch it.